backend-diagram-converter/webapp/src/main/java/org/camunda/community/converter/webapp/ConverterController.java

```diff
@@ -28,64 +30,97 @@
 @RestController
 public class ConverterController {
   private final BpmnConverterService bpmnConverter;
+  private final boolean engineEnabled;
 
   @Autowired
-  public ConverterController(BpmnConverterService bpmnConverter) {
+  public ConverterController(
+      BpmnConverterService bpmnConverter,
+      @Value("${camunda.bpm.client.enabled}") boolean engineEnabled) {
     this.bpmnConverter = bpmnConverter;
+    this.engineEnabled = engineEnabled;
   }
 
   @PostMapping("/check")
-  public ResponseEntity<?> check(@RequestParam("files") MultipartFile[] bpmnFiles) {
-    List<BpmnDiagramCheckResult> results =
-        Arrays.stream(bpmnFiles)
-            .map(
-                bpmnFile -> {
-                  try (InputStream in = bpmnFile.getInputStream()) {
-                    return bpmnConverter.check(
-                        bpmnFile.getOriginalFilename(), Bpmn.readModelFromStream(in), false);
-                  } catch (IOException e) {
-                    BpmnDiagramCheckResult result = new BpmnDiagramCheckResult();
-                    result.setFilename(bpmnFile.getOriginalFilename());
-                    return result;
-                  }
-                })
-            .collect(Collectors.toList());
-    return ResponseEntity.ok(results);
+  public ResponseEntity<?> check(
+      @RequestParam(value = "files", required = false) MultipartFile[] bpmnFiles) {
+    if (bpmnFiles != null && !engineEnabled) {
+      List<BpmnDiagramCheckResult> results =
+          Arrays.stream(bpmnFiles)
+              .map(
+                  bpmnFile -> {
+                    try (InputStream in = bpmnFile.getInputStream()) {
+                      return bpmnConverter.check(
+                          bpmnFile.getOriginalFilename(), Bpmn.readModelFromStream(in), false);
+                    } catch (IOException e) {
+                      BpmnDiagramCheckResult result = new BpmnDiagramCheckResult();
+                      result.setFilename(bpmnFile.getOriginalFilename());
+                      return result;
+                    }
+                  })
+              .collect(Collectors.toList());
+      return ResponseEntity.ok(results);
+    }
+    if (bpmnFiles == null && engineEnabled) {
+      return ResponseEntity.ok(bpmnConverter.checkFromEngine(false));
```

Comment:
Don't read the file from the engine in the web app, it should be done in the cli module.